packages/portal/src/components/item/ItemSnippetCopyButton.vue

```diff
@@ -0,0 +1,181 @@
+<template>
+  <div v-if="text">
+    <div
+      class="snippet-wrapper position-relative mb-1"
+    >
+      <b-button
+        class="copy-button visually-hidden"
+        @click="copySnippet"
+      >
+        {{ buttonText }}
+      </b-button>
+      <component
+        :is="tag"
+        data-qa="item snippet copy button"
+        class="snippet"
+        @click="copySnippet"
+      >
+        {{ text }}
+      </component>
+    </div>
+    <output
+      :class="snippetCopied ? 'd-inline-flex' : 'visually-hidden'"
+      class="copy-to-clipboard-success align-items-center mb-1"
+      data-qa="item snippet copied message"
+    >
+      <span class="icon-check-circle d-inline-block mr-1" />
+      {{ $t('messages.copyToClipboardSuccess') }}
+    </output>
+    <p
+      v-if="helpText"
+      class="help d-flex align-items-center"
+    >
+      <span class="icon-info-outline d-inline-block mr-1" />
+      {{ helpText }}
+    </p>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'ItemSnippetCopyButton',
```

Comment:
I'm not convinced that we should refer to the whole component a "button" when it does not appear styled as one to the user.
Something more like: `ItemShareableSnippet` might be clearer.

Reply:
I've renamed the component to ShareSnippet. As in the future we might reuse it for non-item features.